### ISSUE TITLE
CardsConfig should create UI notifications instead of throwing Errors in case of unallowed but possible actions.

### ISSUE THREAD

**Issue by gizmo2342** [foundryvtt/foundryvtt#8377]
### What happened?

When trying to drag&drop an already drawn card from a cards deck (CardsConfig) to another hand/pile an Error will be thrown. (foundry.js:17091)

As this is a regular UI action, instead of an error a UI notification should be shown and the action should be rejected without error. 

If the action is prevented by a hooked function (multiple locations), there should also be a UI notification. Players typically do not check the console.

### What ways of accessing Foundry can you encounter this issue in?

- [X] Native App (Electron)
- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Other

### Reproduction Steps

see above

### What core version are you reporting this for?

Version 10 (build 288)

### Bug Checklist

- [ ] The issue occurs while all Modules are disabled

**Comment by Fyorl** [comment]
I don't think it's feasible to display a UI notification if the action was blocked by a hook. It's possible to call `Cards#pass` via API and not just the UI, and we would not want an API call to be generating UI notifications. Instead we would have to change the return value of `Cards#pass` to indicate that it was blocked in some way, which is not something we can really do during stable, so it would have to wait for v11 if at all. The rest of the issue is fine.

**Comment by gizmo2342** [comment]
I agree. Perhaps you can add it to your design guides for future versions.

**Comment by Fyorl** [comment]
I've raised a separate issue at #8443 in order to track it. I will repurpose this issue to reflect just the changes that are being made for the next release.
